dm worker prints too much errors

### What did you do?

Submit a DM job to engine

### What did you expect to see?

The job runs properly or exits with an error message



### What did you see instead?

The openapi returns a normal state, while the dm worker frequently prints error logs
```shell
 ✗ curl -X GET http://127.0.0.1:10240/api/v1/jobs
{"jobs":[{"id":"54dac46a-4101-4ca4-8f67-f1ac76dad092", "type":"DM", "state":"Running", "config": ......
```

```log
[2022/10/28 08:24:33.102 +00:00] [ERROR] [worker.go:314] ["task runs with error"] [job_id=54dac46a-4101-4ca4-8f67-f1ac76dad092] [worker_id=adf8060c-961f-41b5-bab3-4fffb10656a1] [task-id=upstream0] ["error msg"="[{\"ErrCode\":34019,\"ErrClass\":\"load-unit\",\"ErrScope\":\"not-set\",\"ErrLevel\":\"high\",\"RawCause\":\"[Lightning:Restore:ErrCreateSchema]create schema failed, table: `tiflow_test`.``, stmt: restore database schema: run create schema job failed: Error 1273: Unsupported collation when new collation is enabled: 'utf8mb4_0900_ai_ci'\"}]"]
[2022/10/28 08:24:33.102 +00:00] [INFO] [worker.go:320] ["got auto resume strategy"] [job_id=54dac46a-4101-4ca4-8f67-f1ac76dad092] [worker_id=adf8060c-961f-41b5-bab3-4fffb10656a1] [task-id=upstream0] [strategy="resume task makes no sense"]
[2022/10/28 08:24:33.152 +00:00] [ERROR] [worker.go:314] ["task runs with error"] [job_id=54dac46a-4101-4ca4-8f67-f1ac76dad092] [worker_id=adf8060c-961f-41b5-bab3-4fffb10656a1] [task-id=upstream0] ["error msg"="[{\"ErrCode\":34019,\"ErrClass\":\"load-unit\",\"ErrScope\":\"not-set\",\"ErrLevel\":\"high\",\"RawCause\":\"[Lightning:Restore:ErrCreateSchema]create schema failed, table: `tiflow_test`.``, stmt: restore database schema: run create schema job failed: Error 1273: Unsupported collation when new collation is enabled: 'utf8mb4_0900_ai_ci'\"}]"]
[2022/10/28 08:24:33.152 +00:00] [INFO] [worker.go:320] ["got auto resume strategy"] [job_id=54dac46a-4101-4ca4-8f67-f1ac76dad092] [worker_id=adf8060c-961f-41b5-bab3-4fffb10656a1] [task-id=upstream0] [strategy="resume task makes no sense"]
[2022/10/28 08:24:33.202 +00:00] [ERROR] [worker.go:314] ["task runs with error"] [job_id=54dac46a-4101-4ca4-8f67-f1ac76dad092] [worker_id=adf8060c-961f-41b5-bab3-4fffb10656a1] [task-id=upstream0] ["error msg"="[{\"ErrCode\":34019,\"ErrClass\":\"load-unit\",\"ErrScope\":\"not-set\",\"ErrLevel\":\"high\",\"RawCause\":\"[Lightning:Restore:ErrCreateSchema]create schema failed, table: `tiflow_test`.``, stmt: restore database schema: run create schema job failed: Error 1273: Unsupported collation when new collation is enabled: 'utf8mb4_0900_ai_ci'\"}]"]
[2022/10/28 08:24:33.202 +00:00] [INFO] [worker.go:320] ["got auto resume strategy"] [job_id=54dac46a-4101-4ca4-8f67-f1ac76dad092] [worker_id=adf8060c-961f-41b5-bab3-4fffb10656a1] [task-id=upstream0] [strategy="resume task makes no sense"]
[2022/10/28 08:24:33.253 +00:00] [ERROR] [worker.go:314] ["task runs with error"] [job_id=54dac46a-4101-4ca4-8f67-f1ac76dad092] [worker_id=adf8060c-961f-41b5-bab3-4fffb10656a1] [task-id=upstream0] ["error msg"="[{\"ErrCode\":34019,\"ErrClass\":\"load-unit\",\"ErrScope\":\"not-set\",\"ErrLevel\":\"high\",\"RawCause\":\"[Lightning:Restore:ErrCreateSchema]create schema failed, table: `tiflow_test`.``, stmt: restore database schema: run create schema job failed: Error 1273: Unsupported collation when new collation is enabled: 'utf8mb4_0900_ai_ci'\"}]"]
[2022/10/28 08:24:33.253 +00:00] [INFO] [worker.go:320] ["got auto resume strategy"] [job_id=54dac46a-4101-4ca4-8f67-f1ac76dad092] [worker_id=adf8060c-961f-41b5-bab3-4fffb10656a1] [task-id=upstream0] [strategy="resume task makes no sense"]
[2022/10/28 08:24:33.302 +00:00] [ERROR] [worker.go:314] ["task runs with error"] [job_id=54dac46a-4101-4ca4-8f67-f1ac76dad092] [worker_id=adf8060c-961f-41b5-bab3-4fffb10656a1] [task-id=upstream0] ["error msg"="[{\"ErrCode\":34019,\"ErrClass\":\"load-unit\",\"ErrScope\":\"not-set\",\"ErrLevel\":\"high\",\"RawCause\":\"[Lightning:Restore:ErrCreateSchema]create schema failed, table: `tiflow_test`.``, stmt: restore database schema: run create schema job failed: Error 1273: Unsupported collation when new collation is enabled: 'utf8mb4_0900_ai_ci'\"}]"]
[2022/10/28 08:24:33.302 +00:00] [INFO] [worker.go:320] ["got auto resume strategy"] [job_id=54dac46a-4101-4ca4-8f67-f1ac76dad092] [worker_id=adf8060c-961f-41b5-bab3-4fffb10656a1] [task-id=upstream0] [strategy="resume task makes no sense"]
[2022/10/28 08:24:33.352 +00:00] [ERROR] [worker.go:314] ["task runs with error"] [job_id=54dac46a-4101-4ca4-8f67-f1ac76dad092] [worker_id=adf8060c-961f-41b5-bab3-4fffb10656a1] [task-id=upstream0] ["error msg"="[{\"ErrCode\":34019,\"ErrClass\":\"load-unit\",\"ErrScope\":\"not-set\",\"ErrLevel\":\"high\",\"RawCause\":\"[Lightning:Restore:ErrCreateSchema]create schema failed, table: `tiflow_test`.``, stmt: restore database schema: run create schema job failed: Error 1273: Unsupported collation when new collation is enabled: 'utf8mb4_0900_ai_ci'\"}]"]
```

### Versions of the cluster

Dataflow Engine version (run `tiflow version`):

```console
(paste Dataflow Engine version here)
6baa8bb9c84c92b2fbea1232b9dc22ec1d11be19
```

--- a/engine/executor/dm/worker.go
+++ b/engine/executor/dm/worker.go
@@ -319,13 +319,17 @@ func (w *dmWorker) checkAndAutoResume(ctx context.Context) error {
 		return nil
 	}
 
-	w.Logger().Error("task runs with error", zap.String("task-id", w.taskID), zap.Any("error msg", result.Errors))
 	subtaskStage := &pb.SubTaskStatus{
 		Stage:  pb.Stage_Paused,
 		Result: result,
 	}
 	strategy := w.autoResume.CheckResumeSubtask(subtaskStage, dmconfig.DefaultBackoffRollback)
-	w.Logger().Info("got auto resume strategy", zap.String("task-id", w.taskID), zap.Stringer("strategy", strategy))
+
+	previousStage := w.getStage()
+	if stage != previousStage {
+		w.Logger().Error("task runs with error", zap.String("task-id", w.taskID), zap.Any("error msg", result.Errors))
+		w.Logger().Info("got auto resume strategy", zap.String("task-id", w.taskID), zap.Stringer("strategy", strategy))
+	}
 
 	if strategy == worker.ResumeDispatch {
 		w.Logger().Info("dispatch auto resume task", zap.String("task-id", w.taskID))

